Add breakpoint hover effect in gutter

pb03 opened the pull request:
Fixes Issue: #5603, #6627

Here's the Pull Request Doc
https://firefox-dev.tools/debugger.html/docs/pull-requests.html

### Summary of Changes
Add grey hover effect to the breakpoints in gutter

### Test Plan

* Open a JS file.
* Hover over the gutter line numbers.
* Line numbers should be highlighted.
* Empty lines shouldn't be highlighted.
* Existing breakpoints shouldn't be highlighted.

### Screenshots/Videos

#### Before:
![before](https://user-images.githubusercontent.com/21006120/42631268-551b4a1a-85f7-11e8-998e-2763869f553f.gif)

#### After:
![ff](https://user-images.githubusercontent.com/21006120/42631299-7286e212-85f7-11e8-8f58-9cc10d9cf920.gif)

pb03 commented:
This is almost the same PR which was reverted (https://github.com/devtools-html/debugger.html/pull/6626) except it fixes the bug introduced by it.

jasonLaster commented:
Wow! This looks great. Out of curiosity, how does this succeed when the prior approach failed? 

Couple questions:

- Did you test this while scrolling?
- Did you test this when there were less then 10 lines and more than 1000?
- Could you test this approach with a wasm file? I'm paranoid 😉 #6585

jasonLaster commented:
unrelated, but i wonder if this PR would now work now that we've fixed this weird effect 🤔 
https://github.com/devtools-html/debugger.html/pull/6062

jasonLaster commented:
dont worry about the failing test, that is a known intermittent

darkwing commented:
Thank you for giving this a shot @pb03 !  I'm seeing this when I pull down your code:

![width](https://user-images.githubusercontent.com/46655/42639260-345e8f54-85b5-11e8-9edf-3b55047ead78.gif)

I removed the following code which prevented the odd jump:

```css
:not(.empty-line):not(.new-breakpoint)
  > .CodeMirror-gutter-wrapper:hover
  > .CodeMirror-linenumber {
  left: auto !important;
  right: 6px;
  color: var(--theme-body-color);
}
```

![yay](https://user-images.githubusercontent.com/46655/42639568-f016d9ea-85b5-11e8-8f47-3a57e9904931.gif)

Yay!  The last piece is avoiding the explicit `width`, since the line number could be very short or long (wasm).  I love it so far!

darkwing commented:
The other reason not setting an explicit width is important is because code-folding will extend the line number column width in some cases.

pb03 commented:
@jasonLaster I believe previously explicit value of `left` was used to achieve an appropriate arrow size, I just removed it and fixed the shape size using `mask-size`.

@darkwing Yup, wasn't aware that gutter width can vary. 
I've a better solution in mind (without using explicit width). Let me try that and update the PR.

pb03 commented:
![fixed](https://user-images.githubusercontent.com/21006120/42644573-94a2c902-8619-11e8-86b6-0138a2337abe.gif)
It's fixed now!

jasonLaster commented:
Wow! @pb03 i'm learning more MASK magic ✨ in this PR

pb03 commented:
![press](https://user-images.githubusercontent.com/21006120/42644794-122ec09c-861a-11e8-9d9e-974d9a3e7121.gif)
Note: When selecting a breakpoint if there's wide gutter, the blue arrow gets cut.
It's a separate issue and I think it would be a good idea to show it with CSS background, instead of adding it to the DOM.
